### PR TITLE
feat: option to append URL params to authorization URI for `signIn()`

### PR DESCRIPTION
**Pull request by adoublef**
This was mainly to resolve an issue with Auth0 tokens but may useful for other oauth2 providers.

For Auth0 specifically, if an `audience` search parameter isn't provided then the resulting token is an opaque one, with no body.

This means that the token cannot be used in conjunction with the [Auth0 APIs.](https://auth0.com/docs/api) This fix should hopefully fix that for Auth0 tokens (and by extension other providers should they also need additional url parameters to be added).

Example of the access token that is sent back to the end user with the possible patch

### CODE REVIEW

File: src/sign_in.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  isSecure,
-  OAUTH_COOKIE_NAME,
-  redirect,
-  setOAuthSession,
+    COOKIE_BASE,
+    getCookieName,
+    isSecure,
+    OAUTH_COOKIE_NAME,
+    redirect,
+    setOAuthSession,
```
Comment (iuioiua) [inline src/sign_in.ts:9]: Can you please revert these stylistic changes? You may be using a formatter different from [Deno's in-built one](https://deno.land/manual/tools/formatter). Once reverted, you can just run `deno fmt`.

**Review by iuioiua — CHANGES_REQUESTED** [review]
I think this is a nice and reasonable addition. Thanks! Also, can you please add a simple test for this new functionality?

File: src/sign_in.ts (hunk 1 above)
Comment (adoublef) [reply src/sign_in.ts]: ahh didn't even spot that, yes I think it's been reverted now

**Comment by brianknight10** [comment]
Nice! Thanks @adoublef

**Review by iuioiua — APPROVED** [review]
@adoublef, I've made some tweaks. WYDT?

**Comment by adoublef** [comment]
Yes this is great thank you

**Pull request merged**